crowdfunding/projects/serializers.py

```diff
@@ -0,0 +1,156 @@
+from rest_framework import serializers
+
+from .models import Project, Pledge
+# from users.serializers import CustomUserSerializer
+
+class PledgeSerializer(serializers.ModelSerializer): #doing the serializer no manually
+    supporter_public = serializers.SerializerMethodField()
+    class Meta:
+        model = Pledge
+        fields = ['id', 'amount', 'comment', 'anonymous', 'project', 'supporter_private', 'supporter_public'] 
+        read_only_fields = ['id', 'supporter_private']
+        # extra_kwargs = {'supporter_private': {'write_only': True}} #hiding this but to use as a link to users pledges
+        
+    def get_supporter_public(self, obj):
+        if obj.anonymous:
+            return f"mystery gnome enthusiast"
+        else:
+            return obj.supporter_private.username
```

Comment:
oh! 
you've done this!
ahhhh i see!!! 
you didn't need to add supporter_public to the model.

This makes a lot of sense though, i love what you've done!!